HTTP URL's incorrectly handled as local resources

*From @dulimarta [issue](https://github.com/Banno/polymer-2-starter-kit-webpack/issues/4)*

When trying to include an absolute HTTP link, such as a Google Font, this gets compiled as a local resource. An example is when using any *paper-element* that depends on `font-roboto` package.
```
Module not found: Error: 
Can't resolve '${PROJECT_PATH}/bower_components/font-roboto/https:/fonts.googleapis.com/css?family=Roboto....
```

Steps to reproduce error

1. Use webpack adapted Starter Kit: `git clone https://github.com/Banno/polymer-2-starter-kit-webpack.git`
2. Import some paper element `bower install --save paper-item`
3. Include dependency `<link rel="import" href="../bower_components/paper-item/paper-item.html">`
4. Run `webpack`

--- a/README.md
+++ b/README.md
@@ -22,6 +22,7 @@ The loader transforms your components:
  Any ```<link>``` "href" or ```<script>``` "src" that is **not an external link** and does not start with ```~```, ```/```, ```./``` or a series of ```../``` will have ```./``` appended to the beginning of the value. To prevent this change use options ignoreLinks below. 
 
 Example
+
 | `tag:`                            | `import:`                        |
 | ----------------------------------- | ------------------------------------- |
 | `<link rel="import" href="path/to/some-element.html">`     | `import "./path/to/some-element.html"`  |

--- a/README.md
+++ b/README.md
@@ -22,16 +22,13 @@ The loader transforms your components:
  Any ```<link>``` "href" or ```<script>``` "src" that is **not an external link** and does not start with ```~```, ```/```, ```./``` or a series of ```../``` will have ```./``` appended to the beginning of the value. To prevent this change use options ignoreLinks below. 
 
 Example
-
-`<link rel="import" href="path/to/some-element.html">` -> `import "./path/to/some-element.html"`
-
-`<link rel="import" href="/path/to/some-element.html">` -> `import "/path/to/some-element.html"`
-
-`<link rel="import" href="../path/to/some-element.html">` -> `import "../path/to/some-element.html"`
-
-`<link rel="import" href="./path/to/some-element.html">` -> `import "./path/to/some-element.html"`
-
-`<link rel="import" href="~path/to/some-element.html">` -> `import "~path/to/some-element.html"`
+| `tag:`                            | `import:`                        |
+| ----------------------------------- | ------------------------------------- |
+| `<link rel="import" href="path/to/some-element.html">`     | `import "./path/to/some-element.html"`  |
+| `<link rel="import" href="/path/to/some-element.html">`    | `import "/path/to/some-element.html"`   |
+| `<link rel="import" href="../path/to/some-element.html">`  | `import "../path/to/some-element.html"` |
+| `<link rel="import" href="./path/to/some-element.html">`   | `import "./path/to/some-element.html"`  |
+| `<link rel="import" href="~path/to/some-element.html">`    | `import "~path/to/some-element.html"`   |
 
 ## Configuring the Loader
 

--- a/README.md
+++ b/README.md
@@ -21,9 +21,9 @@ The loader transforms your components:
  
  Any ```<link>``` "href" or ```<script>``` "src" that is **not an external link** and does not start with ```~```, ```/```, ```./``` or a series of ```../``` will have ```./``` appended to the beginning of the value. To prevent this change use options ignoreLinks below. 
 
-Example
+## Path Translations
 
-| `tag:`                            | `import:`                        |
+| `tag`                            | `import`                        |
 | ----------------------------------- | ------------------------------------- |
 | `<link rel="import" href="path/to/some-element.html">`     | `import "./path/to/some-element.html"`  |
 | `<link rel="import" href="/path/to/some-element.html">`    | `import "/path/to/some-element.html"`   |

--- a/README.md
+++ b/README.md
@@ -17,6 +17,22 @@ The loader transforms your components:
  * `<script src="./other-script.js"></script>` -> `import './other-script.js';`
  * `<script>/* contents */</script>` -> `/* contents */`
 
+ What does that mean?
+ 
+ Any ```<link>``` "href" or ```<script>``` "src" that is **not an external link** and does not start with ```~```, ```/```, ```./``` or a series of ```../``` will have ```./``` appended to the beginning of the value. To prevent this change use options ignoreLinks below. 
+
+Example
+
+`<link rel="import" href="path/to/some-element.html">` -> `import "./path/to/some-element.html"`
+
+`<link rel="import" href="/path/to/some-element.html">` -> `import "/path/to/some-element.html"`
+
+`<link rel="import" href="../path/to/some-element.html">` -> `import "../path/to/some-element.html"`
+
+`<link rel="import" href="./path/to/some-element.html">` -> `import "./path/to/some-element.html"`
+
+`<link rel="import" href="~path/to/some-element.html">` -> `import "~path/to/some-element.html"`
+
 ## Configuring the Loader
 
 ```javascript

--- a/src/index.js
+++ b/src/index.js
@@ -1,4 +1,3 @@
-import osPath from 'path';
 import url from 'url';
 import { getAttribute, predicates, query, queryAll, remove, removeFakeRootElements } from 'dom5';
 import loaderUtils from 'loader-utils';
@@ -61,7 +60,7 @@ class ProcessHtml {
         const parseLink = url.parse(href);
         const isExternalLink = parseLink.protocol || parseLink.slashes;
         if (ignoreLinks.indexOf(href) < 0 && ignoredFromPartial.length === 0 && !isExternalLink) {
-          source += `\nimport '${path.replace(/\\/g, '\\\\')}';\n`;
+          source += `\nimport '${path}';\n`;
           lineCount += 2;
         }
       }
@@ -158,8 +157,8 @@ RegisterHtmlTemplate.toBody('${minimized.replace(/'/g, "\\'")}');
       if (src) {
         const parseSrc = url.parse(src);
         if (!parseSrc.protocol || !parseSrc.slashes) {
-          const path = osPath.join(osPath.dirname(this.currentFilePath), src);
-          source += `\nimport '${path.replace(/\\/g, '\\\\')}';\n`;
+          const path = ProcessHtml.checkPath(src);
+          source += `\nimport '${path}';\n`;
           lineOffset += 2;
         }
       } else {

--- a/src/index.js
+++ b/src/index.js
@@ -52,14 +52,15 @@ class ProcessHtml {
       if (href) {
         const checkIgnorePaths = ignorePathReWrites.filter(ignorePath => href.indexOf(ignorePath) >= 0);
         if (checkIgnorePaths.length === 0) {
-          path = osPath.join(osPath.dirname(this.currentFilePath), href);
+          path = ProcessHtml.checkPath(href);
         } else {
           path = href;
         }
 
         const ignoredFromPartial = ignoreLinksFromPartialMatches.filter(partial => href.indexOf(partial) >= 0);
-
-        if (ignoreLinks.indexOf(href) < 0 && ignoredFromPartial.length === 0) {
+        const parseLink = url.parse(href);
+        const isExternalLink = parseLink.protocol || parseLink.slashes;
+        if (ignoreLinks.indexOf(href) < 0 && ignoredFromPartial.length === 0 && !isExternalLink) {
           source += `\nimport '${path.replace(/\\/g, '\\\\')}';\n`;
           lineCount += 2;
         }
@@ -93,7 +94,12 @@ class ProcessHtml {
     });
     const links = queryAll(doc, linkPred);
     links.forEach((linkNode) => {
-      remove(linkNode);
+      const href = getAttribute(linkNode, 'href') || '';
+      const parseLink = url.parse(href);
+      const isExternalLink = parseLink.protocol || parseLink.slashes;
+      if (!isExternalLink) {
+        remove(linkNode);
+      }
     });
     const html = domModule ? domModule.parentNode : doc;
     const minimized = minify(parse5.serialize(html), {
@@ -205,6 +211,11 @@ RegisterHtmlTemplate.toBody('${minimized.replace(/'/g, "\\'")}');
       retVal.sourceMap = sourceMapGenerator.toJSON();
     }
     return retVal;
+  }
+
+  static checkPath(path) {
+    const needsAdjusted = /^[A-Za-z]{1}/.test(path);
+    return needsAdjusted ? `./${path}` : path;
   }
 }
 

